ssas/service/public/middleware_test.go

```diff
@@ -0,0 +1,231 @@
+package public
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/go-chi/chi"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+)
+
+var mockHandler http.HandlerFunc = func(w http.ResponseWriter, r *http.Request) {}
+
+type PublicMiddlewareTestSuite struct {
+	suite.Suite
+	server *httptest.Server
+	rr     *httptest.ResponseRecorder
+}
+
+func (s *PublicMiddlewareTestSuite) CreateRouter(handler ...func(http.Handler) http.Handler) http.Handler {
+	router := chi.NewRouter()
+	router.With(handler...).Get("/", func(w http.ResponseWriter, r *http.Request) {
+		_, err := w.Write([]byte("Test router"))
+		if err != nil {
+			log.Fatal(err)
+		}
+	})
+
+	return router
+}
+
+func (s *PublicMiddlewareTestSuite) SetupTest() {
+	s.rr = httptest.NewRecorder()
+}
+
+func (s *PublicMiddlewareTestSuite) TestRequireTokenAuthWithInvalidSignature() {
+	badToken := "eyJhbGciOiJFUzM4NCIsInR5cCI6IkpXVCIsImtpZCI6ImlUcVhYSTB6YkFuSkNLRGFvYmZoa00xZi02ck1TcFRmeVpNUnBfMnRLSTgifQ.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiYWRtaW4iOnRydWUsImlhdCI6MTUxNjIzOTAyMn0.cJOP_w-hBqnyTsBm3T6lOE5WpcHaAkLuQGAs1QO-lg2eWs8yyGW8p9WagGjxgvx7h9X72H7pXmXqej3GdlVbFmhuzj45A9SXDOAHZ7bJXwM1VidcPi7ZcrsMSCtP1hiN"
+
+	testForToken :=
+		func (next http.Handler) http.Handler {
+			return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
In both this test and the next, you are testing token failure. Why are the handlers different? To put it another way, why is it valid to test readRegData() here and not the following test? Any invalid token should result in the same state in the server, no?